FreydCategoriesForCAP/gap/AdditiveClosure.gi

```diff
@@ -1356,6 +1362,256 @@ InstallGlobalFunction( INSTALL_FUNCTIONS_FOR_ADDITIVE_CLOSURE,
         
     fi;
     
+    if HasIsMonoidalCategory( underlying_category ) and IsMonoidalCategory( underlying_category ) then
+        
+        ##
+        AddTensorUnit( category,
+          function( cat )
+            
+            return ObjectConstructor( cat, [ TensorUnit ( UnderlyingCategory( cat ) ) ] );
+            
+          end );
+        
+        ##
+        AddTensorProductOnObjects( category,
+          function( cat, obj_1, obj_2 )
+            local underlying_category, objdat_1, objdat_2, len_1, len_2;
+            
+            underlying_category := UnderlyingCategory( cat );
+            objdat_1 := ObjectDatum( cat, obj_1 );
+            objdat_2 := ObjectDatum( cat, obj_2 );
+            len_1 := Length( objdat_1 );
+            len_2 := Length( objdat_2 );
+            
+            return ObjectConstructor( cat,
+                    List( [ 0 .. len_1 * len_2 - 1 ], i ->
+                      TensorProductOnObjects( underlying_category,
+                        objdat_1[1 + QuoInt( i, len_2 )],
```

Comment:
The existing code in AdditiveClosure uses the convenience `object[i]` instead of `ObjectDatum( object )[i]` (and `morphism[i, j]` instead of `MorphismDatum( morphism )[i][j]`) -> I suggest to keep this consistent.

Reply:
Nice!

Reply:
done